src/pages/404.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+import React from 'react';
+import { i18n } from '../../i18n.config';
+
+const oldPaths = [
+  '/',
+  '/admins/team',
+  '/events/\\d+',
+  '/feedback/new',
+  '/newsletter',
+  '/community'
+].map(p => {
+  if (p === '/') {
+    return new RegExp(`^${p}$`);
+  }
+
+  return new RegExp(`^${p}`);
+});
+
+const useBackRouteCompatibility = () => {
+  React.useEffect(() => {
+    const { pathname } = window.location;
+    if (oldPaths.some(p => p.test(pathname)) && pathname !== '/') {
+      window.location.replace(`/${i18n.defaultLocale}${pathname}`);
+    } else if (pathname === '/') {
+      window.location.replace(`/${i18n.defaultLocale}`);
+    }
+  }, []);
+};
+
+export default function Custom404() {
+  const [isNeedToRedirect, setIsNeedToRedirect] = React.useState(false);
+  useBackRouteCompatibility();
+
+  React.useEffect(() => {
+    setIsNeedToRedirect(oldPaths.some(p => p.test(window.location.pathname)));
+  }, []);
+
+  if (isNeedToRedirect) {
+    return <div id='loader'></div>;
+  }
```

Comment:
Here we are! I tested it a lot and it seems to work properly! well done!

Here a last minor point: we should show something when the page is not found. For example type to get http://localhost:3000/asd
you'll get a blank page

Here a simple layout to show just in case a page is not found

```suggestion
return (
    <div className='container'>
      <div className='content'>
        <h1>Oops!</h1>
        <p>{"Looks like you've stumbled into the void."}</p>
        <svg
          xmlns='http://www.w3.org/2000/svg'
          width='200'
          height='200'
          viewBox='0 0 24 24'
          fill='none'
          stroke='#ffffff'
          strokeWidth='2'
          strokeLinecap='round'
          strokeLinejoin='round'
          className='icon'
        >
          <circle cx='12' cy='12' r='10'></circle>
          <line x1='12' y1='8' x2='12' y2='16'></line>
          <line x1='12' y1='8' x2='12' y2='12'></line>
          <line x1='12' y1='16' x2='12' y2='16'></line>
        </svg>
        <p>{"But don't panic, just click below to get back on track!"}</p>
        <Link href='/'>Go Home</Link>
      </div>
      <style>{`
        .container {
          display: flex;
          align-items: center;
          justify-content: center;
          height: 100vh;
        }
        .content {
          text-align: center;
        }
        .icon {
          width: 100px;
          height: 100px;
          margin: 20px auto;
          display: block;
        }
        h1 {
          font-size: 2em;
          margin-bottom: 0.5em;
        }
        p {
          font-size: 1.2em;
          margin-bottom: 1em;
        }
        a {
          color: #0070f3;
          text-decoration: none;
          font-weight: bold;
        }
        a:hover {
          text-decoration: underline;
        }
      `}</style>
    </div>
```